src/components/Trading/TradingForm.tsx

```diff
@@ -247,25 +295,36 @@ function TradingForm(props: Props) {
         width="100%"
       >
         <HorizontalLayout margin="8px 0">
-          <AssetSelector
```

Comment:
[Minor] Couldn't we add an `inputRef` prop to `AssetSelector` and use the `register()` call that we use for other inputs? It's quite hard to tell what props belong to the `Controller` and which ones belong to the `AssetSelector` now.

Reply:
I now moved the props from the `<Controller>` back to the actual component so that it's clear where they belong. 
I think we should still make use of the `<Controller>` as it fits this use case pretty well.

Besides that I tried your suggested approach with passing an `inputRef` but this caused issues with the value being `undefined` for some renders which caused our `tradePair` hook to fail. Might be my bad but anyways, I think that it's better now.

Reply:
Let's discuss this face-to-face next week. It can wait until then 🙂

Reply:
I retried passing an `inputRef` to `<AssetSelector>` instead of using `<Controller>` but I still didn't manage to get it working.
What I did is:
- I wrap the AssetSelector with React.forwardRef 
- I add `inputRef={ref}` to the TextField of the AssetSelector
- I pass name="secondaryAsset" and `ref={form.register}` to the AssetSelector

But
Although `secondaryAsset` has a value at first (even in the props of the AssetSelector component on first render) it becomes undefined afterwards.
I tried adding the inputRef to the `SelectProps` and `InputProps` of the TextField but it didn't work either.

I can share my WIP code with you @andywer but right now I have no clue how to fix this.

Reply:
It's fine – thx for looking into it again!
After reading the [Controller docs](https://react-hook-form.com/api#Controller) I think we might really need it.